compiler/src/iree/compiler/Codegen/SPIRV/Passes.cpp

```diff
@@ -298,7 +298,12 @@ void addSPIRVBaseVectorizePassPipeline(OpPassManager &pm) {
     nestedModulePM.addNestedPass<func::FuncOp>(
         createGenericVectorizationPass(options));
   }
-  nestedModulePM.addNestedPass<func::FuncOp>(createSPIRVVectorLoweringPass());
+  nestedModulePM.addNestedPass<func::FuncOp>(
+      createSPIRVInitialVectorLoweringPass());
+  nestedModulePM.addNestedPass<func::FuncOp>(
+      createHoistRedundantVectorTransfersPass());
+  nestedModulePM.addNestedPass<func::FuncOp>(
+      createSPIRVFinalVectorLoweringPass());
```

Comment:
Given that these three passes seem to always be called in sequence it might be worth grouping them in their own function.

Reply:
Good call. Done.